### PR TITLE
add attachedcluater e2e test

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:
add attachedcluater e2e test
**Which issue(s) this PR fixes**:
a part of #436

### CODE REVIEW

File: hack/e2e-test/install.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This script installs Kurator cluster-operater and fleet-manager.
+
+KUBECONFIG_PATH=${KUBECONFIG_PATH:-"${HOME}/.kube"}
+MAIN_KUBECONFIG=${MAIN_KUBECONFIG:-"${KUBECONFIG_PATH}/kurator-host.config"}
+export KUBECONFIG=${MAIN_KUBECONFIG}
+VERSION=${VERSION:-"0.95.27"}
```
Comment (hzxuzhonghu) [inline hack/e2e-test/install.sh:12]: Please change to using the commit id

**Review by hzxuzhonghu — COMMENTED** [review]
Dis you test it?

File: hack/e2e-test/fleet-clusters/e2e-run.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+"$DIR"/attachedcluster.sh
```
Comment (hzxuzhonghu) [inline hack/e2e-test/fleet-clusters/e2e-run.sh:8]: We can use go code to write it

**Comment by LiZhenCheng9527** [comment]
You can see the details of the e2e test in the CI of this PR.

File: e2e/attachedcluster_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		namespace         string
+		fleetname         string
+		memberClusterName string
+		kubeconfigPath    string
+		secret            *corev1.Secret
+		attachedcluster   *clusterv1a1.AttachedCluster
+	)
+
+	ginkgo.BeforeEach(func() {
+		namespace = "default"
```
Comment (hzxuzhonghu) [inline e2e/attachedcluster_test.go:43]: nit: use a random namespace for each test

File: e2e/attachedcluster_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		time.Sleep(5 * time.Second)
+		// step 3.create fleet
+		clusters := []*corev1.ObjectReference{
+			{
+				Name: memberClusterName,
+				Kind: "AttachedCluster",
+			},
+		}
+		fleet := resources.NewFleet(namespace, fleetname, clusters)
+		fleetCreateErr := resources.CreateFleet(kuratorClient, fleet)
```
Comment (hzxuzhonghu) [inline e2e/attachedcluster_test.go:83]: don't we need to check fleet status?

File: e2e/attachedcluster_test.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+
+	"kurator.dev/kurator/e2e/resources"
+	clusterv1a1 "kurator.dev/kurator/pkg/apis/cluster/v1alpha1"
+)
+
+var _ = ginkgo.Describe("[AttachedClusters] AttachedClusters testing", func() {
```
Comment (hzxuzhonghu) [inline e2e/attachedcluster_test.go:32]: we need to cleanup the resources created per test

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**